packages/waku/src/router/client.ts

```diff
@@ -110,11 +119,42 @@ const RouterContext = createContext<{
   prefetchRoute: PrefetchRoute;
 } | null>(null);
 
+const RouteChangeContext = createContext<{
+  unstable_onRouteChangeStartRef: RefObject<ChangeRouteCallback | null>;
+  unstable_onRouteChangeCompleteRef: RefObject<ChangeRouteCallback | null>;
```

Comment:
If we use stable things, we don't need a separate context. Can you put them in RouterContext?

Second request: Let's pass `Set<ChangeRouteCallback>` instead of ref in the context.

Reply:
ok. I made the suggested changes. It changes how the api is, so maybe there will be feedback for that, but the context is gone and sets are used.

should be ready for another review